src/modules/populateList.js

```diff
@@ -0,0 +1,58 @@
+import {
+  tasks, addTask, deleteTask, editTask, saveTasks,
+} from './tasks.js';
+
+const newTaskInput = document.getElementById('new-task');
+const todoList = document.getElementById('todo-list');
+
+const populateList = () => {
+  todoList.innerHTML = '';
+  tasks.forEach((task) => {
+    const taskElement = document.createElement('li');
+    taskElement.innerHTML = `
+      <input type="checkbox" ${task.completed ? 'checked' : ''}>
+      <span>${task.description}</span>
+      <button class="edit"><i class="fa-solid fa-pen-to-square"></i></button>
+      <button class="delete"><i class="fa-solid fa-trash"></i></button>
+    `;
+    const checkbox = taskElement.querySelector('input[type="checkbox"]');
+    checkbox.addEventListener('change', () => {
+      task.completed = !task.completed;
+      saveTasks();
+    });
+    const deleteButton = taskElement.querySelector('.delete');
+    deleteButton.addEventListener('click', () => {
+      deleteTask(task.index);
+      populateList();
+    });
+    const editButton = taskElement.querySelector('.edit');
+    editButton.addEventListener('click', () => {
+      const newDescription = prompt('Enter a new description:', task.description);
```

Comment:
- [ ] **Optional:** Using Windows functions like alert and prompt are not advisable in professional code as they do not permit async code. They pause the flow of code until they are executed. 

Rather than use a prompt, you could use [a modal popup](https://www.w3schools.com/howto/howto_css_modals.asp) to take the inputs from users.

Reply:
Noted with Thanks!